exporter/opentelemetry-exporter-datadog/tests/test_datadog_exporter.py

```diff
@@ -667,3 +673,50 @@ def test_service_name_fallback(self):
 
         span = datadog_spans[0]
         self.assertEqual(span["service"], "fallback_service_name")
+
+    @unittest.skipUnless(
+        hasattr(os, "fork") and sys.version_info >= (3, 7),
+        "needs *nix and minor version 7 or later",
+    )
+    def test_exports_with_fork(self):
+        # pylint: disable=invalid-name
+        tracer_provider = trace.TracerProvider()
+        tracer = tracer_provider.get_tracer(__name__)
+
+        exporter = MockDatadogSpanExporter()
+
+        span_processor = datadog.DatadogExportSpanProcessor(
+            exporter, schedule_delay_millis=50
+        )
+        tracer_provider.add_span_processor(span_processor)
+        with tracer.start_as_current_span("foo"):
+            pass
+        time.sleep(0.5)  # give some time for the exporter to upload spans
+
+        self.assertTrue(span_processor.force_flush())
+        self.assertEqual(len(get_spans(tracer, exporter, shutdown=False)), 1)
+        exporter.reset()
+
+        def child(conn):
+            def _target():
+                with tracer.start_as_current_span("span") as s:
+                    s.set_attribute("i", "1")
+                    with tracer.start_as_current_span("temp"):
+                        pass
+
+            self.run_with_many_threads(_target, 100)
+
+            time.sleep(0.5)
+
+            spans = get_spans(tracer, exporter, shutdown=False)
+            conn.send(len(spans) == 200)
+            conn.close()
+
+        parent_conn, child_conn = multiprocessing.Pipe()
+        fork_context = multiprocessing.get_context("fork")
```

Comment:
I'm running this code on macOS where [the default is `spawn`](https://github.com/open-telemetry/opentelemetry-python/pull/2242). With that default I got a 'Can't pickle local object' when the created `Process` linked to this inner `child` function. I didn't really want to move out `child` but figured this small detail didn't matter much since the test should behave the same on all systems.